zephyr/drivers/can/can_sccan.c

```diff
@@ -264,6 +296,114 @@ static void sc_can_tx_done(const struct device *dev, int status)
 	}
 }
 
+static void sc_can_enable_acceptance_filter(const struct sc_can_cfg *config,
+					     int filter_id, const struct can_filter *filter)
+{
+	uint32_t mask_offset;
+	uint32_t value_offset;
+	uint32_t enable_reg;
+	uint32_t mask_reg = 0;
+	uint32_t value_reg = 0;
+	bool extended = filter->flags & CAN_FILTER_IDE;
+	bool rtr = filter->flags & CAN_FILTER_RTR;
+
+	mask_offset = SCCAN_AFIMR1_OFFSET + (filter_id * 0x20);
+	value_offset = SCCAN_AFIVR1_OFFSET + (filter_id * 0x20);
+
+	/* Enable Acceptance Filter */
+	enable_reg = sc_can_read32(config, SCCAN_AFER_OFFSET);
+	sc_can_write32(config, SCCAN_AFER_OFFSET, enable_reg | BIT(filter_id));
+
+	/* Regist Acceptance Filter Mask and Value */
+	if (filter->flags & CAN_FILTER_IDE) {
+		mask_reg = (SCCAN_AFID_EX1(filter->mask) | SCCAN_AFSRTR(1) | SCCAN_AFIDE(extended) |
+					SCCAN_AFID_EX2(filter->mask) | SCCAN_AFERTR(rtr));
+		value_reg = (SCCAN_AFID_EX1(filter->id) | SCCAN_AFSRTR(1) | SCCAN_AFIDE(extended) |
+					SCCAN_AFID_EX2(filter->id) | SCCAN_AFERTR(rtr));
+	} else {
+		mask_reg = (SCCAN_TXID1(filter->mask) | SCCAN_AFSRTR(rtr) | SCCAN_AFIDE(extended));
+		value_reg = (SCCAN_TXID1(filter->id) | SCCAN_AFSRTR(rtr) | SCCAN_AFIDE(extended));
+	}
+
+	sc_can_write32(config, mask_offset, mask_reg);
+	sc_can_write32(config, value_offset, value_reg);
+
+	return;
+}
+
+static void sc_can_read_idr(const struct sc_can_cfg *config, struct can_frame *can_frame)
+{
+	uint32_t idr;
+
+	idr = sc_can_read32(config, SCCAN_RMR1_OFFSET);
+
+	if (SCCAN_RXIDE(idr)) {
+		can_frame->id = (SCCAN_RXID1_EXT(idr) | SCCAN_RXID2_EXT(idr));
+		can_frame->flags |= CAN_FRAME_IDE;
+		if (SCCAN_RXERTR(idr)) {
```

Comment:
More descriptive name would be appreciated.

Reply:
Changed to:
SCCAN_RX_EXT_FRAME
SCCAN_RX_EXT_REMOTE_FRAME